pkg/validation/validation.go

```diff
@@ -0,0 +1,138 @@
+// Package validation provides UpgradeConfig CR validation tools.
+package validation
+
+import (
+	"fmt"
+	"net/url"
+	"runtime"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/google/uuid"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-version-operator/pkg/cincinnati"
+	upgradev1alpha1 "github.com/openshift/managed-upgrade-operator/pkg/apis/upgrade/v1alpha1"
+	"github.com/openshift/managed-upgrade-operator/pkg/osd_cluster_upgrader"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+)
+
+// NewBuilder returns a validationBuilder object that implements the ValidationBuilder interface.
+func NewBuilder() ValidationBuilder {
+	return &validationBuilder{}
+}
+
+// Validator knows how to validate UpgradeConfig CRs.
+//go:generate mockgen -destination=mockValidation.go -package=validation github.com/openshift/managed-upgrade-operator/pkg/validation Validator
```

Comment:
sure. I did have this there at one stage. I think make squealed because my repo was not clean when I made the change. Resolved: f351af304bf7c65ef25f052469b86f6f430201c5